auditbeat/module/file_integrity/event.go

```diff
@@ -214,6 +215,17 @@ func NewEvent(
 	return NewEventFromFileInfo(path, info, err, action, source, maxFileSize, hashTypes)
 }
 
+func getDriveLetter(path string) string {
+	volume := filepath.VolumeName(path)
```

Comment:
If VolumeName ever returns something that has wide characters this won't work.  Might be safer to range over the string, that way you would know it has 2 runes and that the second rune is == ":"

Reply:
So AFAIK go normalizes normalizes everything to utf8 under the hood in the `filepath` package. Their internal implementation of `VolumeName` assumes everything is utf8, so I think this should be alright here.

Reply:
I'm probably being paranoid, but here is what I'm thinking.

golang strings are made up of runes, runes can be 1 to 4 bytes in length.  If you just use ASCII characters then each rune is 1 byte in length, which is why the current code works.  But if any of the libraries you depend on return a string that contains a rune that takes up more than 1 byte, the code will not do what you expect.  The following is untested but I was thinking of something like this:

``` golang
func getDriveLetter(path string) string {
	volume := filepath.VolumeName(path)
	if utf8.RuneCountInString(volume) != 2 {
		return ""
	}
	last, _ := utf8.DecodeLastRuneInString(volume)
	if last == utf8.RuneError || last != ':' {
		return ""
	}
	first, _ := utf8.DecodeRuneInString(volume)
	if first == utf8.RuneError || !unicode.IsLetter(first) {
		return ""
	}
	return string(unicode.ToUpper(first))
}
```

Reply:
After chatting about this, extracted an `isASCIILetter` function for now and left the utf16 handling for another day, since it doesn't appear to be necessary currently. Left a comment in the new function to specify when we'd need to change the logic for drive-letter mapping in the presence of wide characters.